pol-core/bscript/compiler/analyzer/SemanticAnalyzer.cpp

```diff
@@ -1008,6 +1053,34 @@ void SemanticAnalyzer::visit_identifier( Identifier& node )
   }
 }
 
+void SemanticAnalyzer::visit_index_unpacking( IndexUnpacking& node )
+{
+  if ( auto variable = create_variable( node.source_location, node.scoped_name.scope.string(),
+                                        node.scoped_name.name ) )
+  {
+    node.variable = std::move( variable );
+    visit_children( node );
+  }
+}
+
+void SemanticAnalyzer::visit_member_unpacking( MemberUnpacking& node )
+{
+  if ( node.scoped_name.has_value() )
+  {
+    const auto& scoped_name = node.scoped_name.value();
+    if ( auto variable =
+             create_variable( node.source_location, scoped_name.scope.string(), scoped_name.name ) )
+    {
+      node.variable = std::move( variable );
+    }
+    visit_children( node );
```

Comment:
should be inside the `if`, to only continue analysis if the variable could be defined.